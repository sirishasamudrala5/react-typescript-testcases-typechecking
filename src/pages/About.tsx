import React from 'react';
import { RouteComponentProps } from '@reach/router';
import { useDispatch, useSelector } from 'react-redux';

import { ReduxState } from '../redux/store';
import { incrementOne } from '../redux/actions/sample';

const About: React.FC<RouteComponentProps> = (): JSX.Element => {
  const dispatch = useDispatch();
  const counter = useSelector((state: ReduxState) => state.sample.counter);

  return (
    <section>
      <>
        <h1>About Page</h1>
        <button onClick={() => dispatch(incrementOne)}>Increment</button>
        <p>Current Count: {counter}</p>
      </>
    </section>
  );
};

export default About;
