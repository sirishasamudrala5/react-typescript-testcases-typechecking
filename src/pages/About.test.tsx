import React from 'react';
import userEvent from '@testing-library/user-event';
import { render, screen } from '../../test/test-utils';


import About from './About';

it('renders <About /> page', () => {
  // You should be able to verify the About page rendered correctly.
  const { aboutPage } = render(<About />)
  expect(aboutPage).toMatchSnapshot();
});

it('clicks button and fires increment counter', () => {
  // You should be able to click the increment button and verify the count.
  render(<About />)

  const { inc1 } = render(
      <p>Current Count: 1</p>,
  );
  const { inc2 } = render(
    <p>Current Count: 2</p>,
);
  // Increment
  userEvent.click(screen.getByText('Increment'))
  expect(inc1).toMatchSnapshot();

  // Increment again
  userEvent.click(screen.getByText('Increment'))
  expect(inc2).toMatchSnapshot();
  
});
