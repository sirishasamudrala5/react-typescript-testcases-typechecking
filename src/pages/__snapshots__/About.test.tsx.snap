// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`clicks button and fires increment counter 1`] = `undefined`;

exports[`clicks button and fires increment counter 2`] = `undefined`;

exports[`renders <About /> page 1`] = `undefined`;
